scripts/src/javatests/org/oppia/android/scripts/docs/KDocValidityCheckTest.kt

```diff
@@ -463,25 +696,101 @@ class KDocCheckTest {
       }
       """.trimIndent()
     tempFolder.newFolder(
-      "testfiles", "app", "src", "main", "java", "org", "oppia", "android", "app", "activity"
+      "testfiles", "app", "src", "main", "java", "org", "oppia", "android", "app", "home"
     )
     val exemptedFile = tempFolder.newFile(
-      "testfiles/app/src/main/java/org/oppia/android/app/activity/ActivityComponent.kt"
+      "testfiles/app/src/main/java/org/oppia/android/app/home/HomeActivity.kt"
     )
     exemptedFile.writeText(testContent)
+    val exemptionFile = File("${tempFolder.root}/$pathToProtoBinary")
+    val builder = KDocValidityExemptions.newBuilder()
+    builder.addExemptedFilePath("app/src/main/java/org/oppia/android/app/home/HomeActivity.kt")
+    val exemptions = builder.build()
+    exemptions.writeTo(exemptionFile.outputStream())
 
     runScript()
 
     assertThat(outContent.toString().trim()).isEqualTo(KDOC_CHECK_PASSED_OUTPUT_INDICATOR)
   }
 
+  @Test
+  fun testKDoc_redundantExemption_checkShouldFail() {
```

Comment:
I suggest framing this as "fileWithAllExpectedKdocs_redundantExemption_..." and then add a test for the case where it's exempted and only has some of the kdocs--that shouldn't fail since the file hasn't been fully resolved yet.

Reply:
Done.